os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -580,47 +598,128 @@ void binary_manager_release_binary_sem(int bin_idx)
 	irqrestore(flags);
 }
 
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
 /****************************************************************************
- * Name: binary_manager_read_header
+ * Name: binary_manager_read_common_header
  *
  * Description:
- *	 This function reads header and checks whether it is valid or not.
+ *	 This function reads common binary header and checks whether it is valid or not.
  *
  ****************************************************************************/
-int binary_manager_read_header(char *path, user_binary_header_t *header_data, bool crc_check)
+int binary_manager_read_common_header(char *devpath, common_binary_header_t *header_data, bool crc_check)
 {
 	int fd;
 	int ret;
 	uint32_t read_size;
-	uint32_t file_size;
-	bool need_unlink;
+	uint32_t bin_size;
+	uint32_t crc_value = 0;
+	uint8_t *crc_buffer;
+	uint32_t crc_bufsize;
+
+	memset(header_data, 0, sizeof(common_binary_header_t));
+	crc_buffer = NULL;
+
+
+	fd = open(devpath, O_RDONLY);
+	if (fd < 0) {
+		bmdbg("Failed to open %s: %d, errno %d\n", devpath, ret, errno);
+		return ERROR;
+	}
+
+	/* Read the binary header */
+	ret = read(fd, (FAR uint8_t *)header_data, sizeof(common_binary_header_t));
+	if (ret != sizeof(common_binary_header_t)) {
+		bmdbg("Failed to read %s: %d, errno %d\n", devpath, ret, errno);
+		goto errout_with_fd;
+	}
+
+	/* Verify header data */
+	if (header_data->header_size == 0 || header_data->binary_size == 0 ||\
+		header_data->version < BM_VERSION_DATE_MIN || header_data->version > BM_VERSION_DATE_MAX) {
+		bmdbg("Invalid header data : headersize %u, binsize %u, version %u\n", header_data->header_size, header_data->binary_size, header_data->version);
+		goto errout_with_fd;
+	}
+
+	if (crc_check) {
+		crc_bufsize = 512000;
```

Comment:
What is this size from?
And how about making define instead of direct use?

Reply:
good opinion. I defined and useed CMNLIB_CRC_BUFSIZE.